#! /bin/sh /usr/share/dpatch/dpatch-run
## 75_vtysh__vtysh.c__PAGER.dpatch by  <ch@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: vtysh__vtysh.c__PAGER

@DPATCH@

--- old/vtysh/vtysh.c.orig	2004-10-18 01:23:16.000000000 +0200
+++ new/vtysh/vtysh.c	2004-10-18 01:25:15.000000000 +0200
@@ -240,10 +240,16 @@
 
   pager_defined = getenv ("VTYSH_PAGER");
 
-  if (pager_defined)
+  if (pager_defined) {
     vtysh_pager_name = strdup (pager_defined);
-  else
-    vtysh_pager_name = strdup ("more");
+  } else {
+    struct stat pager_stat;
+    if (stat("/usr/bin/pager", &pager_stat) == 0) {
+      vtysh_pager_name = strdup ("/usr/bin/pager");
+    } else {
+      vtysh_pager_name = strdup ("more");
+    }
+  }
 }
 
 /* Command execution over the vty interface. */
